kernel/src/lib.rs

```diff
@@ -157,10 +157,10 @@ fn init_thread() {
         karg.get_initproc_envp().to_vec(),
     )
     .expect("Run init process failed.");
+
     // Wait till initproc become zombie.
     while !initproc.status().is_zombie() {
-        crate::thread::Thread::yield_now();
-        ostd::cpu::sleep_for_interrupt();
+        ostd::task::halt_for_preempt();
```

Comment:
I suggest renaming `task::halt_for_preempt` to `task::halt_cpu`. I think this name conveys the purpose of this method better. From the user's perspective, all she wants is to _halt the CPU_ to save some CPU cycles. But this can only be done properly when taking into considerations how task scheduling and preemption is done. The fact that this method is a preemption point is more like an implementation detail. So I think keep the purpose in the name, while leaving the details in its Rust doc would be better.

This is just a minor suggestion. It is your call to make the final decision.